wrappers/python/nntile/layer/batch_norm.py

```diff
@@ -0,0 +1,163 @@
+# @copyright (c) 2022-present Skolkovo Institute of Science and Technology
+#                              (Skoltech), Russia. All rights reserved.
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/nntile/layer/layer_norm.py
+# LayerNorm of NNTile Python package
+#
+# @version 1.0.0
+
+from operator import mul
+from functools import reduce
+
+from nntile.tensor import (
+    Tensor, TensorMoments,
+    prod_async, prod_slice_async, pow_async, sum_slice_async, sum_fiber_async, norm_slice_async, 
+    hypot_scalar_inverse_async, fill_async, add_slice_async, add_slice3_async,
+    add_fiber_async, add_scalar_async,
+    hypot_scalar_inverse_async,
+    prod_fiber_async, prod_fiber3_async,
+    add_async,
+
+    zeros, ones
+)
+
+from nntile.layer.base_layer import BaseLayer
+
+class BatchNorm2d(BaseLayer):
+    def __init__(self, x: TensorMoments, weight: Tensor, bias: Tensor, x_res_tm: TensorMoments, x_mean: Tensor, inv_std: Tensor, eps: float = 1e-05, redux: bool = False):
+        self.x_tm = x
+        self.x = x.value
+        self.eps = eps**0.5 # we will square it in forward (implementation constraints)
+
+        # TODO: add usage of learnable transforms 
+        self.weight = weight
+        self.bias = bias
+
+        self.x_res_tm = x_res_tm
+        self.x_res = self.x_res_tm.value
+
+        self.x_normalized_copy = zeros(self.x_res.shape, dtype=type(x.value))
+        self.x_mean = x_mean
+        self.inv_std = inv_std
+        self.std = zeros(inv_std.shape, dtype=type(x.value))
+        
+        self.std_tmp_3dim = zeros(self.x.shape[1:], dtype=type(x.value))
+        self.std_tmp_2dim = zeros([self.x.shape[1]]+self.x.shape[3:], dtype=type(x.value))
+
+        self.n_channels = self.x.shape[1]
+        self.numel_in_channel = reduce(mul, self.x.shape)/self.n_channels
+
+        self.grad = None
+
+        self.redux = False
+
+        self.tmp_buff_full = zeros(self.x.shape, dtype=type(x.value))
+
+        self.grad_x_inv_denominator = zeros(self.x.shape, dtype=type(x.value))
+        
+        self.tmp_buff_channels = zeros([self.x.shape[1]], dtype=type(x.value))
+        self.tmp_buff_channels_internal = zeros([self.x.shape[1]], dtype=type(x.value))
+
+    @staticmethod
+    def generate_simple(x, eps: float = 1e-05, redux=False):
```

Comment:
In general, it is better to use `classmethod`.

Reply:
done